network/src/commonMain/kotlin/com/wire/kalium/network/utils/NetworkUtils.kt

```diff
@@ -0,0 +1,54 @@
+package com.wire.kalium.network.utils
+
+import com.wire.kalium.network.api.ErrorResponse
+import com.wire.kalium.network.exceptions.KaliumException
+import io.ktor.client.call.*
+import io.ktor.client.features.*
+import io.ktor.client.statement.*
+import io.ktor.util.*
+import kotlinx.serialization.SerializationException
+
+sealed class NetworkResponse<out T> {
+    data class Success<out T : Any>(val response: HttpResponse, val body: T) : NetworkResponse<T>()
+    data class Error<out E : KaliumException>(val kException: KaliumException) : NetworkResponse<E>()
+}
+
+fun <T> NetworkResponse<T>.successValue(): T = (this as NetworkResponse.Success).body
+fun <T> NetworkResponse<T>.isSuccessful(): Boolean = this is NetworkResponse.Success
+fun <T> NetworkResponse<T>.asHttpResponseCode(): Int = (this as NetworkResponse.Success).response.status.value
```

Comment:
Maybe a bit more of consistency in the naming? the prefix `as` is applied in here, but not in the `responseHeaders` for example.

Not sure if maintaining or removing would be the best. Personally, I would love to remove all of these casts and force the library user to check the response.